spiceaidocs/content/en/reference/Postgres/_index.md

```diff
@@ -0,0 +1,109 @@
+---
+type: docs
+title: "PostreSQL"
+linkTitle: "PostreSQL"
+description: 'PostgreSQL reference'
+weight: 80
+---
+
+PostgreSQL can be connected to Spice AI as either a dataset source, a data store, or for federated SQL query.
+
+## Dataset Source
+
+To use PostgreSQL as a dataset source, specify `postgres` as the `source` portion of `from` value for your dataset.
+
+```yaml
+datasets:
+  - from: postgres:path.to.my_dataset
+    name: my_dataset
+    acceleration:
+        enabled: true
+```
+
+## Federated SQL Query
+
+To use PostgreSQL for federated SQL query, specify `postgres` as the `source` portion of `from` value for your dataset, without acceleration.
+
+```yaml
+datasets:
+  - from: postgres:path.to.my_dataset
+    name: my_dataset
+```
+
+## Data Store
+
+To use PostgreSQL as a data store, specify `postgres` as the `engine` for your dataset.
+
+```yaml
+datasets:
+  - from: spiceai:path.to.my_dataset
+    name: my_dataset
+    acceleration:
+        engine: postgres
+```
+
+## Configuration
+
+You can configure the connection to your PostgreSQL by providing the following `params`:
+
+- `pg_host`: The hostname of the PostgreSQL server.
+- `pg_port`: The port of the PostgreSQL server.
+- `pg_db`: The name of the database to connect to.
+- `pg_user`: The username to connect with.
+- `pg_pass_key`: The name of the secret containing the password to connect with.
```

Comment:
```suggestion
- `pg_pass_key`: The name of the secret containing the password to connect with.
```

This is technically the name of the `key` within the `secret` using our nomenclature. The secret is the key-value object.